src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,92 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { FormEvent, useEffect, useState } from 'react';
+import { getTodos, processRequest } from '../../api/todos';
+import { TodoForm } from '../TodoForm/TodoForm';
+import { Todo } from '../../types/Todo';
+import { TodoList } from '../TodoList/TodoList';
+import { TodoFooter } from '../TodoFooter/TodoFooter';
+import { ErrorNotification } from '../ErrorNotification/ErrorNotification';
+import { Errors } from '../../enums/Errors';
+import { FilteredTodos } from '../../enums/FilteredTodo';
+
+const handleFilteredTodos = (todos: Todo[], filterSelected: FilteredTodos) => {
+  switch (filterSelected) {
+    case FilteredTodos.active:
+      return todos.filter(todo => !todo.completed);
+    case FilteredTodos.completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [filterSelected, setFilterSelected] = useState<FilteredTodos>(
+    FilteredTodos.all,
+  );
+
+  const preparedTodos = handleFilteredTodos(todos, filterSelected);
+  const activeTodos = handleFilteredTodos(todos, FilteredTodos.active);
+  const completedTodos = handleFilteredTodos(todos, FilteredTodos.completed);
+
+  const clearErrorMessage = () => {
+    setErrorMessage(null);
+  };
+
+  const showError = (error: Errors) => {
+    setErrorMessage(error);
+
+    setTimeout(() => {
+      clearErrorMessage();
+    }, 3000);
+  };
+
+  useEffect(() => {
+    processRequest(getTodos, clearErrorMessage)
```

Comment:
If you need to clear an error message just do it and then send regular request with `getTodos` function